chronos/streams/chunkstream.nim

```diff
@@ -68,161 +87,116 @@ proc setChunkSize(buffer: var openarray[byte], length: int64): int =
       i = i - 4
     buffer[c] = byte(0x0D)
     buffer[c + 1] = byte(0x0A)
-    result = c + 2
+    c + 2
 
 proc chunkedReadLoop(stream: AsyncStreamReader) {.async.} =
   var rstream = cast[ChunkedStreamReader](stream)
   var buffer = newSeq[byte](1024)
   rstream.state = AsyncStreamState.Running
 
-  try:
-    while true:
+  while true:
+    try:
       # Reading chunk size
-      var ruFut1 = awaitne rstream.rsource.readUntil(addr buffer[0], 1024, CRLF)
-      if ruFut1.failed():
-        rstream.error = ruFut1.error
-        rstream.state = AsyncStreamState.Error
-        break
-
-      let length = ruFut1.read()
-      var chunksize = getChunkSize(buffer.toOpenArray(0,
-                                                      length - len(CRLF) - 1))
+      let res = await rstream.rsource.readUntil(addr buffer[0], 1024, CRLF)
+      var chunksize = getChunkSize(buffer.toOpenArray(0, res - len(CRLF) - 1))
 
       if chunksize == 0xFFFF_FFFF_FFFF_FFFF'u64:
```

Comment:
This error value could be a named constant or even better `getChunkSize` could return an `Option` or `Result`.